### PR TITLE
fix(validation): make gate/constraint commands portable and fix Windows quoting edge case

### PR DESCRIPTION
**Pull request by ichinya**
## Summary

This PR improves command execution for validation gates and constraints by removing shell dependency and making behavior consistent across platforms.

### What changed

- Switched gate and constraint command execution to `executable + args` parsing (no `sh -c`).
- Added explicit rejection of shell-only syntax (`&&`, `||`, pipes, redirects, etc.).
- Improved failure reporting to distinguish:
  - unsupported syntax
  - parse errors
  - spawn errors
  - non-zero exit codes
- Updated CLI/docs/schema descriptions to reflect the new portable command format.
- Updated tests to use portable commands instead of shell built-ins.
- Fixed a Windows parsing edge case where a quoted argument ending with a trailing backslash (e.g. `"C:\tmp\"`) was incorrectly treated as an unmatched quote.
- Added regression tests for the trailing-backslash quoted-argument case in both parsers.

## Why

The previous behavior relied on shell execution, which was:
- non-portable
- harder to reason about
- more fragile for quoting/escaping
- less safe by default

This change makes command behavior explicit and predictable on both Unix and Windows.

## Validation

- `cargo clippy -- -D warnings`
- `cargo fmt -- --check`
- Added/ran parser regression tests for trailing backslash in quoted arguments.
- Verified `hlv gates run` and `hlv constraints check` with:
  - `cmd /C exit /B 0 "C:\\tmp\\"` (Windows case)

### CODE REVIEW

File: docs/SPECS.md [hunk 1]
```diff
@@ -532,7 +532,7 @@ rules:
     statement: "Secrets must not appear in logs"
     enforcement:
       - log_policy_check
-    check_command: "grep -rn 'secret\\|password' src/ && exit 1 || exit 0"
+    check_command: "cargo clippy --all-targets -- -D warnings"
```
Comment (lee-to) [inline docs/SPECS.md]: Not everyone has cargo

**Comment by ichinya** [comment]
Good catch, agreed.

I replaced the example in `docs/SPECS.md` with an ecosystem-agnostic, project-specific executable command:
`policy-check --rule no-secrets-in-logs`.

The intent of this section is to show portable `program + args` format, not require Rust tooling.

**Comment by lee-to** [comment]
### BLOCKER: ~300 lines of duplicated parser code

  The command parser is **fully copy-pasted** between `src/cmd/gates.rs` and `src/check/constraints.rs`:

  - `QuoteMode` enum — duplicated
  - `split_command_line()` — duplicated
  - `find_unsupported_shell_syntax()` — duplicated
  - `has_unescaped_double_quote()` — duplicated
  - `ParsedGateCommand` / `ParsedCheckCommand` — identical structs, different names
  - `GateCommandParseError` / `CheckCommandParseError` — identical enums, different names

  Should be extracted into a shared module (e.g. `src/util/command_parser.rs`). The problem already manifested in commit   2 — the trailing backslash fix had to be patched in both files independently.

  ---

  ### Minor issues

  **1. `.gitignore` changes are unrelated to this PR**

  The added `/.* ` rule hides all root-level dotfiles from git tracking. This could silently break tracking of   `.editorconfig`, `.rustfmt.toml`, etc. Should be a separate commit or PR.

  **2. `file_exists_check_command()` test helper still uses `sh -c`**

  `tests/constraints_tests.rs` — the helper wraps commands in `sh -c` on Unix, which contradicts the purpose of this PR.   Since `test` is a standalone executable (`/usr/bin/test`), `test -f marker.txt` can be used directly without the   shell wrapper.

  **3. `traceability.yaml` fixture changes not explained**

  Contracts and tests arrays were emptied in the example fixture. This is likely needed to fix unrelated check failures,   but the PR description doesn't mention it. Worth a note to avoid confusion.

  ---

  ### Nits

  **Trailing backslash heuristic is fragile.** `has_unescaped_double_quote()` uses lookahead to disambiguate `\"` as   escape vs. backslash + closing quote. Breaks on edge cases like `"C:\tmp\" "next arg"` where the lookahead finds the   quote from the next argument. Should be documented as a known limitation.

  **`$VAR` is not blocked.** `find_unsupported_shell_syntax()` rejects `$(` but allows `$HOME` or `${HOME}`. These won't   expand without a shell, but users may be surprised the variable name is passed literally. Intentional or oversight?

**Comment by ichinya** [comment]
Thanks for the thorough review - great catches.
 - I pushed a follow-up commit (`1403f24`) that addresses the blocker and the noted minors.
 - Extracted the duplicated parser into a shared module: `src/util/command_parser.rs`.
 - Switched both `gates` and `constraints` to use the shared parser.
 - Added regression coverage for the trailing-backslash + next-quoted-arg case.
 - Removed the `sh -c` wrapper from the Unix test helper.
 - Reverted the broad `/.*` ignore rule to explicit dot-directories.
 - Restored the traceability fixture mappings and kept docs/schemas aligned with current parser behavior.
 - `make lint` and targeted parser/integration checks pass locally.
If you have time, please take another look.
